### PR TITLE
Add buffer size parameter to CopyTo method

### PR DESCRIPTION
**Pull request by pakrym**
We want to stop using FileStreams buffering, but we'd like to use bigger buffer then 4k for static files.

### CODE REVIEW

**Comment by pakrym** [comment]
cc @davidfowl @rynowak @pranavkm

File: src/Microsoft.AspNetCore.Http.Extensions/StreamCopyOperation.cs [hunk 1]
```diff
@@ -15,7 +15,12 @@ public static class StreamCopyOperation
     {
         private const int DefaultBufferSize = 4096;
 
-        public static async Task CopyToAsync(Stream source, Stream destination, long? count, CancellationToken cancel)
+        public static Task CopyToAsync(Stream source, Stream destination, long? count, CancellationToken cancel)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Http.Extensions/StreamCopyOperation.cs]: Docs while you're editing this file.

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Http.Extensions/StreamCopyOperation.cs [hunk 2]
```diff
@@ -15,7 +15,12 @@ public static class StreamCopyOperation
...
+        {
+            return CopyToAsync(source, destination, count, DefaultBufferSize, cancel);
+        }
+
+        public static async Task CopyToAsync(Stream source, Stream destination, long? count, int bufferSize, CancellationToken cancel)
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Http.Extensions/StreamCopyOperation.cs]: you missed supplying this buffer size to the Rent call

File: src/Microsoft.AspNetCore.Http.Extensions/StreamCopyOperation.cs [hunk 3]
```diff
@@ -15,11 +15,29 @@ public static class StreamCopyOperation
...
         private const int DefaultBufferSize = 4096;
 
-        public static async Task CopyToAsync(Stream source, Stream destination, long? count, CancellationToken cancel)
+        /// <summary>Asynchronously reads the bytes from the source stream and writes them to another stream.</summary>
+        /// <returns>A task that represents the asynchronous copy operation.</returns>
+        /// <param name="source">The stream from which the contents will be copied.</param>
+        /// <param name="destination">The stream to which the contents of the current stream will be copied.</param>
+        /// <param name="count">The count of bytes to be copied.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="P:System.Threading.CancellationToken.None" />.</param>
+        public static Task CopyToAsync(Stream source, Stream destination, long? count, CancellationToken cancellationToken)
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Http.Extensions/StreamCopyOperation.cs]: Renaming the parameter is a source breaking change. I don't think that's allowed for 1.1.0

**Comment by pakrym** [comment]
:up::date:

**Comment by kichalla** [comment]
:shipit:

**Pull request merged**